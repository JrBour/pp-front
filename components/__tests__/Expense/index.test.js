import { shallowMount } from '@vue/test-utils'
import Expense from '@/components/Expense'

describe('Expense', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Expense, {
      propsData: {
        expense: {
          name: 'Name',
          user: {
            givenName: 'John',
            lastName: 'Doe'
          },
          amount: 12.12,
          createdAt: '12/03/2021'
        },
        canRemoveExpense: true
      }
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
