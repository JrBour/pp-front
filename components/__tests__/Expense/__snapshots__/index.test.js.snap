// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense snapshot renders properly 1`] = `
<div class="expense__wrapper"><button>
    <p>x</p>
  </button>
  <div class="expense__date">
    <p class="expense__date_day">Ven</p>
    <p class="expense__date_date">3</p>
    <p class="expense__date_month">Dec</p>
  </div>
  <div class="expense__separator"></div>
  <div class="expense__informations_wrapper">
    <div class="expense__informations">
      <h3>Name</h3>
      <p>John Doe</p>
    </div>
    <div class="expense__amount">
      <p>12.12 â‚¬</p>
    </div>
  </div>
</div>
`;
